web/libs/datamanager/src/components/DataManager/DataManager.jsx

```diff
@@ -66,32 +67,56 @@ const TabsSwitch = switchInjector(
   observer(({ sdk, views, tabs, selectedKey }) => {
     const editable = sdk.tabControls;
 
+    const onDragEnd = useCallback((result) => {
+      if (!result.destination) {
+        return;
+      }
+
+      views.updateViewOrder(result.source.index, result.destination.index);
+    }, []);
+
     return (
       <Tabs
         activeTab={selectedKey}
         onAdd={() => views.addView({ reload: false })}
         onChange={(key) => views.setSelected(key)}
+        onDragEnd={onDragEnd}
         tabBarExtraContent={<ProjectSummary />}
         addIcon={<LSPlus />}
         allowedActions={editable}
       >
-        {tabs.map((tab) => (
-          <TabsItem
-            key={tab.key}
-            tab={tab.key}
-            title={tab.title}
-            onFinishEditing={(title) => {
-              tab.setTitle(title);
-              tab.save();
-            }}
-            onDuplicate={() => tab.parent.duplicateView(tab)}
-            onClose={() => tab.parent.deleteView(tab)}
-            onSave={() => tab.virtual && tab.saveVirtual()}
-            active={tab.key === selectedKey}
-            editable={tab.editable}
-            deletable={tab.deletable}
-            virtual={tab.virtual}
-          />
+        {tabs.map((tab, index) => (
+          <Draggable key={tab.key} draggableId={tab.key} index={index}>
+            {(provided, snapshot) => (
+              <Elem
+                name={"draggable"}
+                ref={provided.innerRef}
+                {...provided.draggableProps}
+                {...provided.dragHandleProps}
+                style={{
+                  background: snapshot.isDragging && "#ddd",
+                  ...provided.draggableProps.style,
+                }}
+              >
+                <TabsItem
+                  key={tab.key}
+                  tab={tab.key}
+                  title={tab.title}
+                  onFinishEditing={(title) => {
+                    tab.setTitle(title);
+                    tab.save();
+                  }}
+                  onDuplicate={() => views.duplicateView(tab)}
```

Comment:
so it works like this? even if the `tab` is potentially removed from the tree?
what about closing tab which was moved? and can we also apply this change to `onClose()`? so it will be `views.deleteView(tab)`

Reply:
We always have the tab on the tree, even if we move the position and reorder the array, the problem that we were having is that for some reason we were loosing the `tab.parent` reference, and because of this, we couldn't get the duplicateView method from views. Yeah make totally sense, I already applied the same to onClose(), thanks @hlomzik